### PR TITLE
stop using CRD v1beta1 in tests

### PR DESCRIPTION
**Pull request by deads2k**
Remove unnecessary v1beta1 dependencies from tests.  This has a few we cannot yet remove, so I have to tweak those tests so they still test something valuable.  I'll remove WIP once I have done that.

/kind cleanup
/priority important-soon
@kubernetes/sig-api-machinery-pr-reviews 

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go [hunk 1]
```diff
@@ -664,58 +638,64 @@ func TestForbiddenFieldsInSchema(t *testing.T) {
...
-			if err := fixtures.DeleteCustomResourceDefinition(noxuDefinition, apiExtensionClient); err != nil {
+			if err := fixtures.DeleteV1CustomResourceDefinition(noxuDefinition, apiExtensionClient); err != nil {
 				t.Fatal(err)
 			}
 		}
 	}
 }
 
 func TestNonStructuralSchemaConditionUpdate(t *testing.T) {
+	t.Skip("non-structural schemas can no longer be created")
```
Comment (deads2k) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go:698]: This test has to be restored and use the v1beta1 client.

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go [hunk 2]
```diff
@@ -862,6 +842,7 @@ spec:
 }
 
 func TestNonStructuralSchemaCondition(t *testing.T) {
+	t.Skip("non-structural schemas can no longer be created")
```
Comment (deads2k) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go:845]: This test has to be restored and use the v1beta1 client.

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go [hunk 3]
```diff
@@ -664,58 +638,64 @@ func TestForbiddenFieldsInSchema(t *testing.T) {
...
 			}
-			validationSchema.OpenAPIV3Schema.AdditionalProperties.Allows = false
-
-			_, err = fixtures.CreateNewCustomResourceDefinition(noxuDefinition, apiExtensionClient, dynamicClient)
-			if err == nil {
-				t.Fatalf("unexpected non-error: additionalProperties cannot be set to false")
-			}
-
-			validationSchema.OpenAPIV3Schema.Properties["zeta"] = apiextensionsv1beta1.JSONSchemaProps{
+			// v1 doesn't allow additional properties at all. this isn't necessary in v1.
```
Comment (deads2k) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go:650]: This needs to be put back to v1beta1 until 1.22, when it can be removed because it is testing client behavior, not etcd/controller behavior.

File: test/integration/apiserver/apply/apply_crd_beta_test.go [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/client-go/dynamic"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+	apiservertesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
+	"k8s.io/kubernetes/test/integration/framework"
+)
+
+// TestApplyCRDNoSchema tests that CRDs and CRs can both be applied to with a PATCH request with the apply content type
+// when there is no validation field provided.
+func TestApplyCRDNoSchema(t *testing.T) {
```
Comment (deads2k) [inline test/integration/apiserver/apply/apply_crd_beta_test.go:42]: This is a straight move.

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/basic_test.go [hunk 5]
```diff
@@ -25,7 +25,8 @@ import (
 	"testing"
 	"time"
 
-	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/basic_test.go:29]: nit: no empty line

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go [hunk 6]
```diff
@@ -335,7 +336,7 @@ func validateNonTrivialConverted(t *testing.T, ctc *conversionTestContext) {
 			client := ctc.versionedClient(ns, createVersion.Name)
 
 			fixture := newConversionMultiVersionFixture(ns, name, createVersion.Name)
-			if !*ctc.crd.Spec.PreserveUnknownFields {
+			if createVersion.Schema.OpenAPIV3Schema.XPreserveUnknownFields == nil || !*createVersion.Schema.OpenAPIV3Schema.XPreserveUnknownFields {
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go:339]: not sure about this change. Do we have fixtures that make use of this schema field? Note that we also still have `crd.Spec.PreserveUnknownFields` in v1, settable through v1beta1 conversion.

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/limit_test.go [hunk 7]
```diff
@@ -22,9 +22,10 @@ import (
 	"strings"
 	"testing"
 
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/limit_test.go:26]: order

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/listtype_test.go [hunk 8]
```diff
@@ -22,37 +22,38 @@ import (
 	"testing"
 	"time"
 
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/listtype_test.go:25]: order

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/pruning_test.go [hunk 9]
```diff
@@ -41,27 +43,37 @@ import (
...
+			{
+				Name:    "v1beta1",
+				Served:  true,
+				Storage: true,
+				Subresources: &apiextensionsv1.CustomResourceSubresources{
+					Status: &apiextensionsv1.CustomResourceSubresourceStatus{},
+				},
+				Schema: &apiextensionsv1.CustomResourceValidation{
+					OpenAPIV3Schema: &apiextensionsv1.JSONSchemaProps{
+						XPreserveUnknownFields: pointer.BoolPtr(false),
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/pruning_test.go:65]: no need to set this to false

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/scope_test.go [hunk 10]
```diff
@@ -21,6 +21,8 @@ import (
 	"strings"
 	"testing"
 
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/scope_test.go:25]: order

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/subresources_test.go [hunk 11]
```diff
@@ -424,21 +389,16 @@ func TestValidationSchemaWithStatus(t *testing.T) {
...
 
 	// fields other than properties in root schema are not allowed
-	noxuDefinition := newNoxuValidationCRDs(apiextensionsv1beta1.NamespaceScoped)[0]
-	noxuDefinition.Spec.Subresources = &apiextensionsv1beta1.CustomResourceSubresources{
-		Status: &apiextensionsv1beta1.CustomResourceSubresourceStatus{},
-	}
-	_, err = fixtures.CreateNewCustomResourceDefinition(noxuDefinition, apiExtensionClient, dynamicClient)
-	if err == nil {
-		t.Fatalf(`unexpected non-error, expected: must not have "additionalProperties" at the root of the schema if the status subresource is enabled`)
-	}
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/subresources_test.go:434]: why is this removed?

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go (hunk 3 above)
Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go:650]: we support it, but not at the root.

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/subresources_test.go (hunk 11 above)
Comment (deads2k) [reply staging/src/k8s.io/apiextensions-apiserver/test/integration/subresources_test.go:434]: > why is this removed?

a side effect of trying to make the schema eventually creatable.  Some tests relied on having a single schema for all versions.  Others wanted two versions.

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/apiapproval_test.go [hunk 12]
```diff
@@ -21,6 +21,8 @@ import (
 	"testing"
 	"time"
 
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/apiapproval_test.go:25]: nit: empty line

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go [hunk 13]
```diff
@@ -27,6 +27,8 @@ import (
 	"testing"
 	"time"
 
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go:30]: order

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go (hunk 6 above)
Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go:339]: We don't. Check `multiVersionFixture`: it is static and has `PreserveUnknownFields` set to false, i.e. the if always applies. Just do the same here: remove the if and leave the `SetNestedField` with `foo: garbage`.

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go [hunk 14]
```diff
@@ -393,7 +394,7 @@ func validateNonTrivialConvertedList(t *testing.T, ctc *conversionTestContext) {
 		name := "converted-" + createVersion.Name
 		client := ctc.versionedClient(ns, createVersion.Name)
 		fixture := newConversionMultiVersionFixture(ns, name, createVersion.Name)
-		if !*ctc.crd.Spec.PreserveUnknownFields {
+		if createVersion.Schema.OpenAPIV3Schema.XPreserveUnknownFields == nil || !*createVersion.Schema.OpenAPIV3Schema.XPreserveUnknownFields {
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go:397]: same here, remove the whole check and lieve the `SetNestedField` (unconditionally)

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go [hunk 15]
```diff
@@ -428,13 +429,12 @@ func validateNonTrivialConvertedList(t *testing.T, ctc *conversionTestContext) {
...
-	if *ctc.crd.Spec.PreserveUnknownFields {
-		return
-	}
-
 	ns := ctc.namespace
 
 	for _, createVersion := range ctc.crd.Spec.Versions {
+		if createVersion.Schema.OpenAPIV3Schema.XPreserveUnknownFields == nil || !*createVersion.Schema.OpenAPIV3Schema.XPreserveUnknownFields {
+			continue
+		}
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go:437]: also here. I don't think this is used at all.

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/finalization_test.go [hunk 16]
```diff
@@ -22,9 +22,7 @@ import (
 	"time"
 
 	"github.com/stretchr/testify/require"
-
-	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
-
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/finalization_test.go:25]: order

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/subresources_test.go (hunk 11 above)
Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/test/integration/subresources_test.go:434]: It amazes me that we have an invalid schema in a standard fixture.

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go [hunk 17]
```diff
@@ -123,81 +120,69 @@ func newNoxuValidationCRDs(scope apiextensionsv1beta1.ResourceScope) []*apiexten
...
-					{
-						Type:      "string",
-						MaxLength: int64Ptr(5),
-					},
-					{
-						Type:    "number",
-						Minimum: float64Ptr(0),
-					},
-				},
-			},
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go:138]: Impossible to follow these removals.

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/subresources_test.go (hunk 11 above)
Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/test/integration/subresources_test.go:434]: ok, found it: for structural schemas we have this:

```
	if g.AdditionalProperties != nil {
		if lvl == rootLevel {
			allErrs = append(allErrs, field.Forbidden(fldPath.Child("additionalProperties"), "must not be used at the root"))
		}
```

In other words, we forbid `additionalProperties` even for CRDs without status subresource. So removing this extra test is fine. Please also remove the now invalid comment above the removal.

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/subresources_test.go [hunk 18]
```diff
@@ -489,24 +454,18 @@ func TestValidateOnlyStatus(t *testing.T) {
...
-			noxuDefinition.Spec.Versions[1].Schema = &apiextensionsv1beta1.CustomResourceValidation{
-				OpenAPIV3Schema: schemaWithDescription,
-			}
+	noxuDefinitions := NewNoxuSubresourcesCRDs(apiextensionsv1.NamespaceScoped)
+	for _, noxuDefinition := range noxuDefinitions {
+		noxuDefinition.Spec.Versions[0].Schema = &apiextensionsv1.CustomResourceValidation{
+			OpenAPIV3Schema: schema,
+		}
+		schemaWithDescription := schema.DeepCopy()
+		schemaWithDescription.Description = "test"
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/subresources_test.go:463]: I have no idea why we had set the decription at all. Would completely remove that `schemaWithDescription` code here and see whether it's green.

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go (hunk 17 above)
Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go:138]: I remember you asked me on slack. Yes, this is not structural. If you have removed the delta test code as well, I think we are fine.

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go [hunk 19]
```diff
@@ -530,18 +515,6 @@ func TestCustomResourceValidationErrors(t *testing.T) {
...
-				instanceFn: func() *unstructured.Unstructured {
-					instance := newNoxuValidationInstance(ns, "foo")
-					instance.Object["delta"] = "foobarbaz"
-					return instance
-				},
-				expectedErrors: []string{
-					"must validate at least one schema (anyOf)",
-					"delta in body should be at most 5 chars long",
-				},
-			},
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go:544]: ack, here it is.

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go (hunk 3 above)
Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go:650]: as this to the schema instead:
```
validationSchema.OpenAPIV3Schema.Properties["eta"]: {
  type: object
  additionalProperties: false
}
```

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go [hunk 20]
```diff
@@ -664,58 +638,64 @@ func TestForbiddenFieldsInSchema(t *testing.T) {
...
 				Type:        "array",
 				UniqueItems: true,
 			}
-			validationSchema.OpenAPIV3Schema.AdditionalProperties.Allows = true
 
-			_, err = fixtures.CreateNewCustomResourceDefinition(noxuDefinition, apiExtensionClient, dynamicClient)
+			// v1 doesn't allow additional properties at all. this isn't necessary in v1.
+			//validationSchema.OpenAPIV3Schema.AdditionalProperties.Allows = true
+
+			_, err = fixtures.CreateNewV1CustomResourceDefinition(noxuDefinition, apiExtensionClient, dynamicClient)
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go:665]: do this instead:
```
validationSchema.OpenAPIV3Schema.Properties["eta"]: {
  type: object
  additionalProperties:
    type: string
}
```

File: test/integration/apiserver/apiserver_test.go [hunk 21]
```diff
@@ -32,9 +32,10 @@ import (
 	"testing"
 	"time"
 
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
```
Comment (sttts) [inline test/integration/apiserver/apiserver_test.go:35]: order

**Comment by deads2k** [comment]
@sttts pushed commits addressing your comments.  Brought back the two v1beta1 validation tests.  One I wired directly against etcd. The second one I've left as an exercise for you to unwind.  I suggest splitting into "things still valid in v1" and "the rest". Then we have to remove "the rest" when we branch master.
